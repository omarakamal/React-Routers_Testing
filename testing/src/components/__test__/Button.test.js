import React from "react";
import ReactDOM from "react-dom"
import Button from "../Button";
import {render, cleanup} from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import renderer from "react-test-renderer"

afterEach(cleanup)
it("renders without crashing",()=>{
    const div=document.createElement("div")
    ReactDOM.render(<Button></Button>,div)
    
})


it("renders button correctly",()=>{
    const {getByTestId} = render(<Button person="Omar"></Button>)
    expect(getByTestId("button")).toHaveTextContent("Omar")
})

it("renders button correctly",()=>{
    const {getByTestId} = render(<Button person="OmarH"></Button>)
    expect(getByTestId("button")).toHaveTextContent("Omar")
})

it("matches snapshot",()=>{
    const tree= renderer.create(<Button label="hi"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
})
