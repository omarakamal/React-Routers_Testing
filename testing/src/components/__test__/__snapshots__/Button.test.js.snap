// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <button
    className="Button-style"
    data-testid="button"
  >
    hid 
  </button>
</div>
`;
